src/SBarcodeScanner.cpp

```diff
@@ -23,14 +24,20 @@ SBarcodeScanner::~SBarcodeScanner()
     stopCam();
 }
 
-SBarcodeDecoder *SBarcodeScanner::getDecoder()
+SBarcodeDecoder* SBarcodeScanner::getDecoder()
 {
     return &m_decoder;
 }
 
-void SBarcodeScanner::initCam() {
+void SBarcodeScanner::initCam()
+{
     camera = new QCamera(this);
 
+    if (camera->error()) {
+        qDebug() << "Error during camera initialization: " << camera->errorString();
```

Comment:
Property `cameraAvaliable` is not set anywhere. Would be best if it'd be set here.